### ISSUE TITLE
SWC parse error when showing detailed view in rtneuronApp

### ISSUE THREAD

**Issue by alex4200** [BlueBrain/RTNeuron#25]
On Ubuntu 18.04.1:

I am unable to see the detailed view. I cloned the whole sonata repository to be sure to have the up-to-date files, but when trying to open the 'Detailed' view from `300_cells` I get the following error messages: 

````
 12258 Unable to load data for neuron a2: Reading swc morphology file: /home/alex/Documents/AppImage/sonata/examples/shared_components/morphologies/Scnn1a_473845048_m.swc, parse error at line 4
 12258 Unable to load data for neuron a3: Reading swc morphology file: /home/alex/Documents/AppImage/sonata/examples/shared_components/morphologies/Rorb_325404214_m.swc, parse error at line 4
 12259 Unable to load data for neuron a4: Reading swc morphology file: /home/alex/Documents/AppImage/sonata/examples/shared_components/morphologies/Rorb_325404214_m.swc, parse error at line 4
```

**Comment by hernando** [comment]
I cannot reproduce this error, but I have a plausible explanation. I remember seeing that when a Qt application is created and the user locale is Spanish, the behavior of all the family of functions for converting strings into numbers changes and the decimal point becomes ','. Assuming your locale is German, you are probably experiencing the same issue. What makes it puzzling is that the conversion functions don't behave differently if Qt is not initialized.
Try starting the AppImage prepending `LANG=C` in the command line and let me know if this error persists. If the problem is gone, we may considering enforcing the C locale (which is probably fine because internationalization was never considered in the design).

**Comment by hernando** [comment]
If the problem is what I think, PR #31 addresses this issue enforcing LANG=C

**Comment by alex4200** [comment]
I tried the workaround, but it does not change anything as far as I can tell. 

Also, my locale should be English and not German.

**Comment by hernando** [comment]
I have tried to reproduce this inside a Docker container with Ubuntu 18.04 and a fresh build of Brion (which is where the error is coming from). I also tried creating a Qt application before parsing to see if it could influence on strtod, but no success. Since I intend to upgrade my system to 18.04 anyway, I will try again with the whole thing, but I think the error is coming from something in you environment.
Can you post the output of the command `env`?

**Comment by hernando** [comment]
As I was suspecting, the problem is in the locale. You have `LC_NUMERIC=fr_FR.UTF-8`, which is not German, but is French, so same problem. I have been able to reproduce your problem exporting that variable (and installing the locale library for glibc). I can also confirm that LANG=C doesn't solve anything. Try exporting `LC_NUMERIC=C`, that should fix it.
I will apply the same fix to the AppImage and #31 instead of using LANG right now. Good that we followed up on this.
